### DISCUSSION TITLE
mark addition failed

### DISCUSSION THREAD

**Discussion by lvhaoyu39** [katspaugh/wavesurfer.js#3149] [Q&A] [answered]
I added a mark (zero-length regions) and reported the following error.
Uncaught TypeError: Cannot read properties of null (reading 'scrollWidth')
    at regions.js:1:6810
    at Proxy.filter (<anonymous>)
    at Proxy.avoidOverlapping (regions.js:1:6709)
    at Proxy.saveRegion (regions.js:1:7061)
    at Proxy.addRegion (regions.js:1:7740)

**Comment by katspaugh** [comment]
What code did you use to add a a mark?

**Reply by lvhaoyu39** [reply]
wavesurferState.wsRegion.addRegion({
    start: currentTime,
    content: `${currentTime}`,
    color: 'hsla(400, 100%, 30%, 0.5)'
  })

**Reply by katspaugh** [reply]
What's inside `${currentTime}`? A number?

**Reply by lvhaoyu39** [reply]
It is a string.

**Reply by lvhaoyu39** [reply]
The entire process involves creating a tag, deleting it, and then reporting that error when creating the tag.

**Reply by lvhaoyu39** [reply]
May I ask what is causing it.

**Reply by katspaugh** [reply]
Looks like the same bug as you already reported here: https://github.com/katspaugh/wavesurfer.js/discussions/3130

**Reply by lvhaoyu39** [reply]
The problem is that after deleting the tag, the deleted data will still be obtained. This is because after deletion, there is no way to add the tag again.

**Reply by lvhaoyu39** [reply]
The current issue is very confusing. I can easily handle the previous issue in the code.

**Reply by katspaugh** [reply]
I cannot reproduce this bug. Please create a minimal example that demonstrates it and post it on https://gist.github.com.

**Reply by lvhaoyu39** [reply]
This is a simple example that will reproduce the bug above.
https://gist.github.com/18310035303/3956cfbfee895259ec1675fbd84c9b30

**Reply by katspaugh** [reply]
By minimal example, I meant something that actually works. And runs on https://wavesurfer-js.org/examples/.

**Reply by lvhaoyu39** [reply]
I have updated the test code and can directly copy it to run. I have found an issue that may occur when using it in Vue.

**Reply by lvhaoyu39** [reply]
Then I found out that the previous [3130](https://github.com/katspaugh/wavesurfer.js/discussions/3130) issue was also related to Vue, and I obtained the area data in the test code normally.

**Reply by katspaugh** [reply]
Yeah, that's what I thought as well, some state-related problem with your Vue app. Please close both discussions if it's solved. Cheers.

**Reply by lvhaoyu39** [reply, marked as answer]
I have solved these two problems and thank you very much for answering them for me. In Vue, reactive or ref cannot be used to declare instances, as this can easily lead to such issues.

**Reply by lvhaoyu39** [reply]
I have another small question. Can't we set the time for playback by playing (startTime, endTime) now?
